### DISCUSSION TITLE
RequestTimeout for a single controller action

### DISCUSSION THREAD

**Discussion by lixaotec** [dotnet/aspnetcore#24205] [General] [answered]
Dear folks, 

How to increse requesttimeout (aspnetcore default is 2 minutes) to a single controller action? 

Is there any way instead of higher in web.config for the whole app?

Thanks in advance

**Comment by Tratcher** [comment, marked as answer]
You must be using IIS out-of-proc. That acts as a proxy so it does not have any granular config.

If you move to IIS in-proc (2.2+) then there will be no default timeout.

**Reply by lixaotec** [reply]
Aspnetcore 2.1 and iis 10 windows server 2016.

Well so it is a good practice, not having a timeout request at prod?

**Reply by Tratcher** [reply]
The trick with timeouts is how you enforce them. System.Web used to do Thread.Abort if the timeout expired. AspNetCore is highly async and doesn't attempt such things. That IIS out-of-proc timeout would close the connection if the request took too long, but it can't actually abort the request processing, that's up to your app code. Your app is in control, you can apply timeouts to any operations you expect to take a non-trivial amount of time. The server applies limits to things that can be directly influenced by the client like request and response min data speeds.

**Reply by lixaotec** [reply]
Thanks @Tratcher . I'll take a better reading on these info.
